### DISCUSSION TITLE
Disable resource save button while there are unfinished file uploads

### DISCUSSION THREAD

**Discussion by jellycow** [filamentphp/filament#211] [Ideas]
I've just encountered an issue when dealing with image uploads in a form. It is possible to save the item updates while an image/file is still being uploaded, this results in the image field data not being saved as it's obviously not finalised. Would it be a good idea/possible to disable the form save button while file/image uploads are still in progress so no data is accidentally lost?

**Comment by zltn** [comment]
+1 on the idea of temporary disabling the save action or any other kind of notification. On top of that I also found myself occasionally forgetting to press save after upload. Especially on resources where I only had to upload a file.

**Comment by loranger** [comment]
I would like to have this feature too.
My users keeps creating entries while files still uploading…

**Comment by danharrin** [comment]
Fixed by #2057, form submit buttons will now be disabled while files are uploading, and will show the "Uploading file..." text instead!

**Reply by loranger** [reply]
Awesome ! Thanks 👌🏻

**Reply by fritzfr** [reply]
This doesn't seem to work for me (am on v3). Can you explain how a form has to be wired up (or the Actions) such that they will be disabled? @danharrin

**Reply by edwink75** [reply]
Doesn't work for me either.
This seems to be for Version 2 :-), I'm trying in version 3. Have the same issue with multiple uploads.
There is a js file that generated the preview images, I think we can use some events to listen to and then disable the button
vendor/filament/forms/resources/js/components/file-upload.js

**Comment by edwink75** [comment]
```
        var total_files = 0

        document.addEventListener('FilePond:initfile', (e) => {
            total_files++;
            toggleButton();
        });

        document.addEventListener('FilePond:processfile', (e) => {
            total_files--;
            toggleButton();
        });

        function toggleButton() {
            var button = document.getElementById('submit_button');
            if (total_files > 0) {
                button.disabled = true;
            } else {
                button.disabled = false;
            }
        }
```
Since the FileUpload component is based on FilePondJs https://pqina.nl/filepond/docs/api/instance/events/
we can use the browser events that the library dispatches.

Not sure if above code is the best solution possible, but it seems to be working fine :-)

**Comment by zohaSZ** [comment]
> I've just encountered an issue when dealing with image uploads in a form. It is possible to save the item updates while an image/file is still being uploaded, this results in the image field data not being saved as it's obviously not finalised. Would it be a good idea/possible to disable the form save button while file/image uploads are still in progress so no data is accidentally lost?

For anyone still encountering this issue, I've added the following to the **FileUpload** field:
```
->extraAttributes([
    'x-on:change' => '$wire.set("isUploading", true)',
])
```
This sets the tracking variable isUploading to true immediately when a file is selected. Then reset it to false in afterStateUpdated:
```
->afterStateUpdated(function ($state) {
    $this->isUploading = false;
})

```
Finally, use these states on your button to enable/disable it:

```
<x-filament::button 
    type="submit"
    :disabled="$this->isUploading"
>
    Send Notification
</x-filament::button>
```
